src/hotspot/cpu/aarch64/c2_MacroAssembler_aarch64.cpp

```diff
@@ -297,53 +297,57 @@ void C2_MacroAssembler::fast_lock_lightweight(Register obj, Register box, Regist
       cmp(zr, obj);
       b(slow_path);
     } else {
+      if (UseObjectMonitorTable) {
```

Comment:
`aarch64` always did a cache lookup even when `!UseObjectMonitorTable` which always results in a slow path.

Reply:
I see that now.  Thanks for removing OMUseC2Cache.